massa-grpc/src/error.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+use std::error::Error;
+
+use displaydoc::Display;
+
+use massa_consensus_exports::error::ConsensusError;
+use massa_execution_exports::ExecutionError;
+use massa_hash::MassaHashError;
+use massa_models::error::ModelsError;
+use massa_network_exports::NetworkError;
+use massa_protocol_exports::ProtocolError;
+use massa_time::TimeError;
+use massa_wallet::WalletError;
+
+/// Errors of the gRPC component.
+#[non_exhaustive]
+#[derive(Display, thiserror::Error, Debug)]
+pub enum GrpcError {
+    /// `massa_hash` error: {0}
+    MassaHashError(#[from] MassaHashError),
+    /// consensus error: {0}
+    ConsensusError(#[from] ConsensusError),
+    /// execution error: {0}
+    ExecutionError(#[from] ExecutionError),
+    /// Network error: {0}
+    NetworkError(#[from] NetworkError),
+    /// Protocol error: {0}
+    ProtocolError(#[from] ProtocolError),
+    /// Reflection error : {0}
+    ReflectionError(#[from] tonic_reflection::server::Error),
+    /// Models error: {0}
+    ModelsError(#[from] ModelsError),
+    /// Time error: {0}
+    TimeError(#[from] TimeError),
+    /// Wallet error: {0}
+    WalletError(#[from] WalletError),
+    /// Internal server error: {0}
+    InternalServerError(String),
+}
+
+impl From<GrpcError> for tonic::Status {
+    fn from(error: GrpcError) -> Self {
+        match error {
```

Comment:
Does it make sense for all these errors to be mapped to internal errors? Maybe the status codes could be more explicit for some of them? https://github.com/grpc/grpc/blob/master/doc/statuscodes.md#status-codes-and-their-use-in-grpc
However I haven't checked when all these errors are thrown, maybe it's appropriate

Reply:
It's not that rich as HTTP status. By default it's an internal error.